From: win7-support-bot <win7@cromite.org>
Date: Sun, 26 Jul 2026 00:00:00 +0000
Subject: WIN7: ui-views / com

Files:
 ui/views/win/hwnd_message_handler.h

License: GPL-2.0-or-later - https://spdx.org/licenses/GPL-2.0-or-later.html
---
diff --git a/ui/views/win/hwnd_message_handler.h b/ui/views/win/hwnd_message_handler.h
index 3c371d3e624a1..deba19101c4e7 100644
--- a/ui/views/win/hwnd_message_handler.h
+++ b/ui/views/win/hwnd_message_handler.h
@@ -269,6 +269,19 @@ class VIEWS_EXPORT HWNDMessageHandler : public gfx::WindowImpl,
   bool HasSystemFrame() const;
 
  private:
+// Returns true if the window is arranged via Snap. For example, the browser
+// window is snapped via buttons shown when the mouse is hovered over window
+// maximize button.
+static bool IsWindowArranged(HWND window) {
+  // IsWindowArranged() is not a part of any header file.
+  // https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-iswindowarranged
+  using IsWindowArrangedFuncType = BOOL(WINAPI*)(HWND);
+  static const auto is_window_arranged_func =
+      reinterpret_cast<IsWindowArrangedFuncType>(
+          base::win::GetUser32FunctionPointer("IsWindowArranged"));
+  return is_window_arranged_func ? is_window_arranged_func(window) : false;
+}
+
   friend class ::views::test::DesktopWindowTreeHostWinTestApi;
 
   using TouchIDs = std::set<DWORD>;

